### PR TITLE
[metal] Enable offline cache by default on Metal

### PR DESCRIPTION
**Pull request by PGZXB**
Issue: #6263, #4401

### CODE REVIEW

**Review by ailzhang — APPROVED** [review]
Thanks!

File: taichi/runtime/program_impls/metal/metal_program.cpp [hunk 1]
```diff
@@ -136,10 +136,7 @@ const std::unique_ptr<metal::CacheManager>
     TI_ASSERT(compiled_runtime_module_.has_value());
     using Mgr = metal::CacheManager;
     Mgr::Params params;
-    params.mode =
-        offline_cache::enabled_wip_offline_cache(config->offline_cache)
```
Comment (lin-hitonami) [inline taichi/runtime/program_impls/metal/metal_program.cpp:140]: Should we delete function `enabled_wip_offline_cache` ?

**Review by lin-hitonami — APPROVED** [review]
LGTM!

**Pull request merged**